ui/library/src/utils/MaxVisibleElement.ts

```diff
@@ -0,0 +1,17 @@
+import { Nullable } from '../components/types/utils';
+
+type VisibleElementKey = Nullable<number | string>;
+
+export function getMaxVisibleElement(
+  visibleElements: Map<VisibleElementKey, number>
+): VisibleElementKey {
+  let maxVisibleKey = null;
+  let maxRatio = 0;
+  for (const [visibleKey, ratio] of visibleElements) {
+    if (maxRatio < ratio) {
+      maxVisibleKey = visibleKey;
+      maxRatio = ratio;
+    }
+  }
+  return maxVisibleKey;
+}
```

Comment:
Good to extract out the logic. However, the files under `utils` folder has inconsistent naming logic. Some of them use lower camel cases while others use upper. For this file, I think there should be a better name. Or you can rename it as `getMax.ts` since getting max visible element also lies under the category of `getMax`.

Reply:
@yensung Paul was telling us to use the uppercase naming instead of lowercase so i started with it

Reply:
I don't have strong opinions about using the uppercase or lowercase except consistency. I think it's better to rename utility files and make their naming rule consistent. Can you create a follow-up tech debt ticket to address it?